src/ds/features/auto-exposure-gain-limit-feature.h

```diff
@@ -0,0 +1,46 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+
+#include <src/feature-interface.h>
+
+#include <memory>
+
+namespace librealsense {
+
+class synthetic_sensor;
+class hw_monitor;
+
+class range_limit_feature : public feature_interface
```

Comment:
Not exactly what I meant. As it is now, there is no need for `range_limit_feature`, it serves no purpose.

option 1 - Having 2 completely separate features for AE and gain.
Delete `range_limit_feature` and split `auto_exposure_limit_feature` and `gain_limit_feature` into separate files.

option 2 - Using `range_limit_feature` for common implementation.
Delete the public interface, we won't have objects of this type.
For instance you can have protected API with functions like:
    `option_range get_range( rs2_option opt );`
    `std::shared_ptr< limits_option > create_limits_option( rs2_option opt, const char * description );`
    `...`
Or each inheriting feature will set members that a common function will use.

Your call which option you prefer